gnovm/pkg/gnolang/type_check.go

```diff
@@ -215,6 +215,151 @@ func assertAssignableTo(n Node, xt, dt Type, autoNative bool) {
 	}
 }
 
+func assertValidConstExpr(store Store, last BlockNode, n *ValueDecl, expr Expr) {
```

Comment:
can you add some more godoc information? like saying that this checks for const expr validity according to go's spec?